### PR TITLE
RD2: Refactor Validation Service

### PR DESCRIPTION
**Pull request by andrewyu-salesforce**
Allow RD2_ValidationService to run independently outside trigger context.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by andrewyu-salesforce** [comment]
Attach To [W-8247722](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008h74OIAQ/view)

**Comment by andrewyu-salesforce** [comment]
@lparrott @force2b @npsp-reedestockton The code is ready for review.

File: src/classes/RD2_DatabaseService.cls [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Recurring Donations
+* @description Database DML service to insert/update/delete records and keep track of failed RD Ids.
+* Use without sharing context so Guest Site User can perform DML operation
+*/
+public without sharing class RD2_DatabaseService {
```
[the remark this answers is not in the document]
Comment (andrewyu-salesforce) [reply src/classes/RD2_DatabaseService.cls:34]: The guest site users and community will access it through screen flow or other custom flow. More detail can be found in this [WI](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008hZbKIAU/view)

**Review by force2b — APPROVED** [review]
Thanks for clarifying the usage @andrewyu-salesforce. I understand the use-case, but I do think that it'll get flagged during a security review. My recommendation is to add some additional comments to the top of the class to further clarify exactly why it's needed for the "guest site user". There's good content in the work item too, which an also just be referenced by number in the comment block. That'll help if we're asked to justify this in the future.

**Review by npsp-reedestockton — APPROVED** [review]
Looks great, @andrewyu-salesforce !!  Nice work!

**Pull request merged**